vividus-plugin-rest-api/src/main/java/org/vividus/http/PublishingAttachmentInterceptor.java

```diff
@@ -53,16 +55,18 @@ public void process(HttpRequest request, EntityDetails entityDetails, HttpContex
     {
         byte[] body = null;
         String mimeType = null;
+        boolean isBinaryContent = false;
         if (request instanceof HttpEntityContainer httpEntityContainer)
         {
             HttpEntity entity = httpEntityContainer.getEntity();
             if (entity != null)
             {
-                mimeType = MimeTypeUtils.getMimeTypeFromHeaders(request.getHeaders())
+                mimeType = ContentTypeHeaderParser.getMimeTypeFromHeaders(request.getHeaders())
                         .orElseGet(() ->
                                 Optional.ofNullable(ContentType.parseLenient(entity.getContentType()))
                                         .orElse(ContentType.DEFAULT_TEXT).getMimeType()
                         );
+                isBinaryContent = entity instanceof ByteArrayEntity;
```

Comment:
it's not always true: 
 - if mimetype is textual, then it's not binary (even if it's of `ByteArrayEntity`)
 - if mimetype is not textual or not present, then we _assume_ the content is binary, if  it's not of `StringEntity`